### DISCUSSION TITLE
Kami Dump... then?

### DISCUSSION THREAD

**Discussion by rpkilgore** [roleoroleo/yi-hack-Allwinner-v2#237] [Q&A] [answered]
So, , I have a kami mini might also be called the Kami Home Ai.  
Version from the App is 9.20.xxxx
I used the backup script and was able to get the files  for the home and rootfs.  Actually the dump gave ne what looks like everything from the camera.
![image](https://user-images.githubusercontent.com/19828182/129233661-bc283871-2ad3-4b3d-9c3f-7a40f7dcd998.png)

I was able to download the repo and compile everything, but my dumb question is:
what files do I have to modify or where do I have to place the  home and rootfs files (stock files) for this to apply the hack for the camera?
and once the files are in the right location, do I just run the pack_fw.all.sh or re-run the compile?

More info:
serial number: IFUSY48I38ZQ4A201119
Was able to use the y21ga and that did get the webpage, telnet, and ssh working, but no rtsp streams or snapshots..

**Comment by roleoroleo** [comment]
Before to build the repo you should run a couple of tasks:

1. Edit scripts/common.sh to add the definition about the new camera.
2. Create a new dir and the new files in the sdhack folder.
3. Create a new dir and the new files in the sysroot folder.
4. Check the format of the fshareframebuffer file and edit everything that is required to run snapshot and rtsp.
5. ...

You don't need to "place" the rootfs files in the tree, you need to copy only the files required in step 2 and step 3.
When the tree is ok you have to run compile.sh and pack_fw.all.sh

Don't use a hack made for a different model. The hack will overwrite your original files with a wrong version.

**Comment by roleoroleo** [comment]
Or share it with me and I will check if it's possible to add it to this repo.

**Reply by rpkilgore** [reply]
Ok, thanks for the info.  It's seems there are so many different versions of the yi hack it's hard to decide witch one to use.  Is the yi hack v5 the newer one or is this one newer?  I like this method as it seems more like it "side" loads the hack instead of using ch_root to modify the rootfs or homefs.  Looks like the yi hack v5 modifies the rootfs..  I might be wrong here just trying to learn :)

Uploading the backup folder for you to look at.  Thanks!  Out of curiosity, do you mount the rootfs and look for something specific?  Again just trying to learn some of the details of the hack since I have 4 other cameras that I eventually want to add this hack to.  1 older yi dome, 1 kami outside bullet cam, 2 newer dome camera's  that look like astronauts lol, and 2 of these Kami home/ai/mini cams..  I currently don't have access to the other camera's to get any info..  Too bad yi didn't leave a backdoor to actually get into the cam remotely otherwise I could really hack the other cams LOL


[backup.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/6995243/backup.zip)

**Reply by roleoroleo** [reply]
> Ok, thanks for the info. It's seems there are so many different versions of the yi hack it's hard to decide witch one to use. Is the yi hack v5 the newer one or is this one newer? I like this method as it seems more like it "side" loads the hack instead of using ch_root to modify the rootfs or homefs. Looks like the yi hack v5 modifies the rootfs.. I might be wrong here just trying to learn :)

v5 is for hisilicon platform, mstar for mstar platform, allwinner for allwinner platform.
They are different hardware and you can't choose the version you want.

> Uploading the backup folder for you to look at. Thanks! Out of curiosity, do you mount the rootfs and look for something specific?

I check the structure of the file system to confirm that the hack is compatible: same folders, same processes, etc...
Normally the difference between two cams is little: sensor, resolution, wifi chipset.
Then I extract some files and I use them to create the hack and the unhack procedures.

**Comment by roleoroleo** [comment, marked as answer]
I already encountered this cam.
See this issue:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/155

**Reply by rpkilgore** [reply]
Thanks for the reply and checking this out for me!  I really appreciate the help and info.

I tried the y28ga_0.1.7.tar.gz from that issue:  everything loads, but the rstp stream gives connection refused and nothing appears for snapshots.

I also installed the y28ga_0.1.8 version, again everything loads except for the rstp stream and the snapshots do nothing.

This is the what i get from vlc
Your input can't be opened:
VLC is unable to open the MRL 'rtsp://192.168.1.211/ch0_2.h264'. Check the log for details.
Your input can't be opened:
VLC is unable to open the MRL 'rtsp://192.168.1.211/ch0_0.h264'. Check the log for details.
When i get the more details it's simply connection refused by peer

Since ssh works are there any logs from the camera I could pull or any information I could get from the camera to help figure out why the streams aren't working?

**Comment by roleoroleo** [comment]
If I'm not wrong, I started the support for this cam but the work is not completed.
To enable rtsp I need 2 or 3 copies of the framebuffer file /dev/fshare_frame_buf to analyze it.
Could you share with me?

- copy the file
- wait 5 secs
- copy the file
- wait 5 secs
- copy the file
- zip
- share

**Reply by rpkilgore** [reply]
I'd love to help you add rstp for this camera!

I'm unable to find that file tho,  there is no fshare_frame_buf dir in /dev and I can't find  framebuffer file
![image](https://user-images.githubusercontent.com/19828182/130258200-7b31e552-10a5-4619-83f1-b313e23c2449.png)

did a little more digging and it looks like the rstp wd service is running.   
 1447 root      1068 S    {wd_rtsp.sh} /bin/sh /tmp/sd/yi-hack/script/wd_rtsp.sh
but it just keeps trying to restart the rRSTPServer. 

 I was able to get the debug output for you if that helps.

[rRTSPserver.log](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/7024223/rRTSPserver.log)

Let me know if there is anything else i can grab for you that might help,

Also, do you want to move this discussion over to the [issue #155](https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/155) so you have it tracked?

I'm going to owe you a few beers after this :)

**Comment by noman1228** [comment]
I'm actually right next to you guys here. I've got the tools and a bit of knowledge LMK if I can help.
WSL but I can blow the dust off a laptop with tux if need be. Got one IFUSY48 Y28GA, Then two IFUSY65 Y211GU all virgin and waiting for solder'd up pads.

**Comment by roleoroleo** [comment]
Yes, move to issue 155, please.
The file is in /dev/shm/
I got confused with another model.

**Reply by rpkilgore** [reply]
Discussion has been moved to issue [#155 ](https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/155)
